src/views/MissionPlanningView.vue

```diff
@@ -607,14 +800,18 @@ const hideContextMenu = (): void => {
 const setHomePosition = async (): Promise<void> => {
   if (!currentCursorGeoCoordinates.value) return
   const newHome: [number, number] = [currentCursorGeoCoordinates.value[0], currentCursorGeoCoordinates.value[1]]
-  home.value = newHome
-  await vehicleStore.setHomeWaypoint(newHome, 0)
-  openSnackbar({
-    variant: 'success',
-    message: `Home position set to ${newHome[0].toFixed(2)}, ${newHome[1].toFixed(2)}`,
-  })
-  if (planningMap.value) {
-    planningMap.value.setView(newHome, zoom.value)
+  try {
+    home.value = newHome
+    await vehicleStore.setHomeWaypoint(newHome, 0)
+    openSnackbar({
+      variant: 'success',
+      message: `Home position set to ${newHome[0].toFixed(2)}, ${newHome[1].toFixed(2)}`,
+    })
+  } catch (error) {
+    openSnackbar({
+      variant: 'error',
+      message: `Failed to set home position: ${error}`,
+    })
```

Comment:
This function had just been created in the previous commit, and is being completely rewritten here.
18 extra lines to be reviewed.

Reply:
@ArturoManzoli I believe this was not fixed.